src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueHelperLogic.sol

```diff
@@ -0,0 +1,111 @@
+import { SharesMathLib } from "src/interfaces/external/Morpho/MorphoBlue/libraries/SharesMathLib.sol";
+import { IOracle } from "src/interfaces/external/Morpho/MorphoBlue/interfaces/IOracle.sol";
+import { UtilsLib } from "src/interfaces/external/Morpho/MorphoBlue/libraries/UtilsLib.sol";
+import { MorphoLib } from "src/interfaces/external/Morpho/MorphoBlue/libraries/periphery/MorphoLib.sol";
+import { console } from "@forge-std/Test.sol";
+
+/**
+ * @title Morpho Blue Helper contract.
+ * @notice Helper implementation including health factor logic used by Morpho Blue Adaptors.
+ * @author 0xEinCodes, crispymangoes
+ * NOTE: helper functions made virtual in case future Morpho Blue Pair versions require different implementation logic.
+ */
+contract MorphoBlueHelperLogic {
+    // Using libraries from Morpho Blue codebase to ensure same mathematic methods
+    using MathLib for uint128;
+    using MathLib for uint256;
+    using UtilsLib for uint256;
+    using SharesMathLib for uint256;
+    using MorphoLib for IMorpho;
+
+    /**
+     * @notice The Morpho Blue contract on current network.
+     */
+    IMorpho public immutable morphoBlue;
+
+    // Constant from Morpho Blue
+    uint256 constant ORACLE_PRICE_SCALE = 1e36;
+
+    constructor(address _morphoBlue) {
+        morphoBlue = IMorpho(_morphoBlue);
+    }
+
+    /**
+     * @notice The ```_getHealthFactor``` function returns the current health factor of a respective position given an exchange rate.
+     * @param _id The specified Morpho Blue market Id.
+     * @param _market The specified Morpho Blue market.
+     * @return currentHF The health factor of the position atm.
+     */
+    function _getHealthFactor(Id _id, MarketParams memory _market) internal view virtual returns (uint256 currentHF) {
+        uint256 borrowAmount = _userBorrowBalance(_id, address(this));
+        if (borrowAmount == 0) return type(uint256).max;
+        uint256 collateralPrice = IOracle(_market.oracle).price(); // recall from IOracle.sol that the units will be 10 ** (36 - collateralUnits + borrowUnits) BUT collateralPrice is in units of borrow.
+
+        // get collateralAmount in borrowAmount for LTV calculations
+        uint256 collateralAmount = _userCollateralBalance(_id, address(this));
+        console.log(
+            "userCollateralBalance: %s, collateralPrice: %s, collateralAmount: %s",
+            _userCollateralBalance(_id, address(this)),
+            collateralPrice,
+            collateralAmount
+        );
+        uint256 collateralAmountInBorrowUnits = collateralAmount.mulDivDown(collateralPrice, ORACLE_PRICE_SCALE);
+
+        if (collateralAmountInBorrowUnits == 0) return 0;
+        uint256 LTV_PRECISION = 1e18;
+        uint256 currentPositionLTV = borrowAmount.mulDivUp(LTV_PRECISION, collateralAmountInBorrowUnits);
+        // convert LTVs to HF
+        currentHF = (_market.lltv).mulDivDown(1e18, currentPositionLTV);
```

Comment:
```suggestion
        currentHF = _market.lltv.mulDivDown(collateralAmountInBorrowUnits, borrowAmount);
```

should be more precise and avoids an early return (complex logic)

Reply:
I think this works, I've added it to the changes. Thank you!